server/routes/api/landing_page.py

```diff
@@ -455,23 +461,28 @@ def data(dcid):
                 del spec_and_stat[category][topic]
             else:
                 spec_and_stat[category][topic] = filtered_charts
-        if not spec_and_stat[category]:
-            del spec_and_stat[category]
+        # Don't delete a category if it is in the valid categories list.
+        if (not spec_and_stat[category]) and ("validCategories"
```

Comment:
It looks like we're only doing del spec_and_stat[category] if not spec_and_stat[category] and category not in raw_page_data["validCategories"][dcid]["category"]. But what if it's not spec_and_stat[category], but there's no "validCategories" in raw_page_data or dcid is not in raw_page_data["validCategories"]? Should we also del in those cases or are those cases not possible?

Reply:
Done